net: openthread: allow to configure platform info

edmont opened the pull request:
This commit adds the option to configure the platform information string of OpenThread.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

jukkar commented on subsys/net/l2/openthread/Kconfig.thread:
```diff
@@ -55,10 +55,14 @@ config OPENTHREAD_POLL_PERIOD
 	int "The maximum number of children"
-	range 10 512
+	range 10 511
 	default 32
 
 config OPENTHREAD_MAX_IP_ADDR_PER_CHILD
-	int "The maximum number of IPv6 address registrations per child."
+	int "The maximum number of IPv6 address registrations per child"
```
Looks like these changes should be in some other PR, as the commit talks only about configuring platform info.

edmont replied:
I updated the commit description.

jukkar replied:
Thanks. The changes look unrelated, could you split the commit to two, so this PR would have two commits for these changes.

edmont replied:
Done.

edmont committed:
net: openthread: allow to configure platform info

This commit adds the option to configure the platform information
string of OpenThread.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont committed:
net: openthread: fix maximum number of children range

This commit corrects the maximum allowed amount of children to
match Thread specification.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
